.github/workflows/deploy.yml

```diff
@@ -23,33 +31,77 @@ jobs:
                   ./setup-db.bash
 
             - name: Install dependencies
-              run: yarn install --frozen-lockfile
-            - name: Build website
+              env:
+                  NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+              run: yarn install
+
+            - name: Run build tasks (which contain benchmark test files)
               env:
                   NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
               run: yarn build
 
-            - name: Upload Build Artifact
-              uses: actions/upload-pages-artifact@v3
+            # We use the name and email for the GitHub Actions bot, as provided via this link: https://api.github.com/users/github-actions%5Bbot%5D
+            - name: Set Git user
+              run: |
+                  git config user.name "github-actions[bot]"
+                  git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+
+            - name: Discard any changes
+              run: git restore .
+
+            - name: Switch branch to gh-pages branch
+              run: git switch gh-pages
+
+            - name: Commit updated build files to gh-pages branch
+              run: |
+                  cp -r packages/docs/build/ ./
+                  git add .
+                  git commit -m "build: updated build files" || echo "Ignored commit failure, proceeding"
```

Comment:
why ignore the commit failure?

Reply:
The `cp` command prior might not always have modified files, so we only need to proceed to the next step if there's no modifications to commit